pkg/kudoctl/cmd/get/get.go

```diff
@@ -3,60 +3,144 @@ package get
+	if err := opts.Output.Validate(); err != nil {
+		return err
+	}
 
 	err := validate(args)
 	if err != nil {
 		return err
 	}
 
-	kc, err := env.GetClient(settings)
+	var objs []runtime.Object
+	switch args[0] {
+	case "instances":
+		objs, err = opts.Client.ListInstances(opts.Namespace)
+	case "operators":
+		objs, err = opts.Client.ListOperators(opts.Namespace)
+	case "operatorversions":
+		objs, err = opts.Client.ListOperatorVersions(opts.Namespace)
+	case "all":
+		return runGetAll(opts)
+	}
 	if err != nil {
-		return fmt.Errorf("creating kudo client: %w", err)
+		return fmt.Errorf("failed to retrieve objects: %v", err)
 	}
 
-	p, err := getInstances(kc, settings)
-	if err != nil {
-		log.Printf("Error: %v", err)
+	if opts.Output != "" {
+		outObj := []interface{}{}
+		for _, o := range objs {
+			outObj = append(outObj, o)
+		}
+		return output.WriteObjects(outObj, opts.Output, opts.Out)
 	}
+
 	tree := treeprint.New()
 
-	for _, plan := range p {
-		tree.AddBranch(plan)
+	metadataAccessor := meta.NewAccessor()
+	for _, obj := range objs {
+		name, err := metadataAccessor.Name(obj)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve name from %v: %v", obj, err)
+		}
+		tree.AddBranch(name)
 	}
-	fmt.Printf("List of current installed instances in namespace \"%s\":\n", settings.Namespace)
-	fmt.Println(tree.String())
+	_, _ = fmt.Fprintf(opts.Out, "List of current installed %s in namespace \"%s\":\n", args[0], opts.Namespace)
+	_, _ = fmt.Fprintln(opts.Out, tree.String())
 	return err
 }
 
-func validate(args []string) error {
-	if len(args) != 1 {
-		return errors.New(`expecting exactly one argument - "instances"`)
+func runGetAll(opts CmdOpts) error {
+	instances, err := opts.Client.ListInstances(opts.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get instances")
+	}
+	operatorversions, err := opts.Client.ListOperatorVersions(opts.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get operatorversions")
+	}
+	operators, err := opts.Client.ListOperators(opts.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get operators")
 	}
 
-	if args[0] != "instances" {
-		return fmt.Errorf(`expecting "instances" and not %q`, args[0])
+	if opts.Output != "" {
+		outObj := []interface{}{}
+		for _, o := range operators {
+			outObj = append(outObj, o)
+		}
+		for _, o := range operatorversions {
+			outObj = append(outObj, o)
+		}
+		for _, o := range instances {
+			outObj = append(outObj, o)
+		}
+		return output.WriteObjects(outObj, opts.Output, opts.Out)
 	}
 
+	return printAllTree(opts, operators, operatorversions, instances)
+}
+
+func printAllTree(opts CmdOpts, operators, operatorversions, instances []runtime.Object) error {
+
+	rootTree := treeprint.New()
+	for _, o := range operators {
+		op, _ := o.(*v1beta1.Operator)
+		opTree := rootTree.AddBranch(op.Name)
+
+		for _, ovo := range operatorversions {
+			ov, _ := ovo.(*v1beta1.OperatorVersion)
+			if ov.Spec.Operator.Name == op.Name {
```

Comment:
It's not really optional - `corev1.ObjectReference` is a struct, and therefore `ov.Spec.Operator.Name` can never be nil. It can be "", but that's a different issue...

Reply:
You're right, `Operator` won't be nil, so this won't crash.